feat(pgadmin): add new module for pgAdmin

AJ0070 opened the pull request:
Closes #36 
claim #36

## Description
This pull request introduces a new Coder module for `pgAdmin`. This module allows users to easily add a pgAdmin web interface to their workspaces for managing PostgreSQL databases.

<!-- Briefly describe what this PR does and why -->
- The `pgAdmin` module under the `AJ0070` namespace.
- A `main.tf` file to define the Coder app and script resources.
- A `run.sh` script to handle the installation and launch of pgAdmin.
- A `README.md` with usage instructions and prerequisites.
- A basic test file `main.test.ts` to ensure the module loads correctly.

## Type of Change

- [x] New module
- [ ] Bug fix
- [ ] Feature/enhancement
- [ ] Documentation
- [ ] Other

## Module Information
**Path:** `registry/AJ0070/modules/pgadmin`
**New version:** `v1.0.0`
**Breaking change:** [ ] Yes [x] No

## Testing & Validation

- [x] Tests pass (`bun test`)
- [x] Code formatted (`bun run fmt`)
- [x] Changes tested locally

/claim #36

AJ0070 commented on registry/AJ0070/modules/pgadmin/main.tf:
```diff
@@ -0,0 +1,44 @@
+  display_name = "pgAdmin"
+  slug         = "pgadmin"
+  icon         = "/icon/postgres.svg"
+  url          = "http://localhost:${var.port}"
```
Let me look into this; meanwhile, can you assign the issue to me?

AJ0070 replied:
fixed this in latest commit: [served over the subdomain by default, and then fallback to localhost](https://github.com/coder/registry/pull/228/commits/0338bd7821dbf9205521ea44f920db5b149a8ce8)

AJ0070 replied:
Thank you

DevelopmentCats commented on registry/AJ0070/modules/pgadmin/README.md:
````diff
@@ -0,0 +1,21 @@
+```tf
+module "pgadmin" {
+  count    = data.coder_workspace.me.start_count
+  source   = "[registry.coder.com/AJ0070/pgadmin/coder](https://registry.coder.com/AJ0070/pgadmin/coder)"
````
Sorry one more thing I notice.

I would change this to only `registry.coder.com/AJ0070/pgadmin/coder`

since this will be be copied by users to use in templates.

AJ0070 replied:
Kindly review the latest commit and let me know if it aligns with your request.

AJ0070 replied:
Sorry for the misunderstanding in the review; it has been corrected now.

DevelopmentCats commented:
I'm not sure if you had the chance to test this after your changes, but pgadmin4 does not seem to load over the subdomain.

I'm making a few changes, and then Ill push my update to your fork

AJ0070 commented:
> @AJ0070 
> 
> I went to test this module. It was having issues with installation, and the app was not being proxied.
> 
> I forked the registry and made a branch to see if I can understand what is going wrong here.
> 
> The main things that I changed on my side to get this working, was I introduced a variable for the pgadmin config, and introduced locals to format the url and healthcheck.
> 
> I then added a step to write the configuration to a file so that pgadmin loads with the ideal settings for coder.
> 
> Here is the fork if you would like to take a look. https://github.com/DevelopmentCats/registry/tree/cat/AJ0070-pgadmin-fix/registry/AJ0070
> 
> 
> 
> 

Thanks for the details!
I'll look into this

AJ0070 commented:
Thank you so much for taking the time to test the module and provide such detailed feedback. I really appreciate you forking the repository and creating a working example it was incredibly helpful for understanding the issues with installation and proxying.
Your solution is much more stable. With your permission, I would love to incorporate the changes from your `cat/AJ0070-pgadmin-fix` branch into my PR to get this merged.

Please let me know if that's okay with you. Thanks again for your help!

DevelopmentCats commented:
> Thank you so much for taking the time to test the module and provide such detailed feedback. I really appreciate you forking the repository and creating a working example it was incredibly helpful for understanding the issues with installation and proxying. Your solution is much more stable. With your permission, I would love to incorporate the changes from your `cat/AJ0070-pgadmin-fix` branch into my PR to get this merged.
> 
> Please let me know @DevelopmentCats if that's okay with you. Thanks again for your help!

No worries.

That's totally fine with me. I'm happy to provide in whatever way I can. Sorry I couldn't point out the exact issues right out of the gate. I had never really worked with pgadmin-web before now and it took me a minute to get it myself lol.

AJ0070 commented:
Thanks for the confirmation!

AJ0070 commented:
I've implemented the changes. Let me know if there's anything else you'd like adjusted.

AJ0070 commented:
> @AJ0070, please link a video demo of the module wokring in a Coder workspace. Thanks,

I'll do that right after I fix the workflow failure.

AJ0070 commented:
https://github.com/user-attachments/assets/2a2bee93-14b2-4a6b-b1c9-55f94c189c3f

AJ0070 commented:
@DevelopmentCats, could you please review the recent changes when you have a moment?

bpmct commented:
Hey @AJ0070. that video doesn't show pgadmin with coder. the agent doesn't start and there is no pgadmin button in the dashboard

AJ0070 commented:
> > Now pgAdmin icon is visible:
> > <img alt="image" width="1220" height="492" src="https://private-user-images.githubusercontent.com/87864439/470335481-39abb409-366e-413e-b71d-4a6692013f67.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMzNjc1MTQsIm5iZiI6MTc1MzM2NzIxNCwicGF0aCI6Ii84Nzg2NDQzOS80NzAzMzU0ODEtMzlhYmI0MDktMzY2ZS00MTNlLWI3MWQtNGE2NjkyMDEzZjY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzI0VDE0MjY1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5NmM4NzUzNzVlMTY4Njk3YzlkMWY0Zjg5ZjRmMWIzOWNmMTM5Y2M2OWU0NDNkODVhYWU3MTIyZDVlMDFkNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XKcfn917NLTDOFQf7nxvyaWGo9dDHY2-8eWLuWWVIKk">
> > but I am not sure why the connection is refusing:
> > ```
> > 2025-07-24 13:47:00.932 [debu]  apphealth: error checking app health  id=9830f5f2-bf28-4ef1-981a-090ccaf65dbb  slug=pgadmin  now_unhealthy=true  error="Get \"http://localhost:5050/\": dial tcp [::1]:5050: connect: connection refused"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @DevelopmentCats Did you encounter anything like this during your local testing? Let me know if you have any suggestions for resolving it.
> 
> Even in my version, it still took forever to build.
> 
> PGAdmin seems to be pretty heavy overall, and I doubt there is much to be done to speed the install up. In my testing though I was able to get pgadmin installed, and available over the coder app button.

Were you able to open it?

DevelopmentCats commented:
> > > Now pgAdmin icon is visible:
> > > <img alt="image" width="1220" height="492" src="https://private-user-images.githubusercontent.com/87864439/470335481-39abb409-366e-413e-b71d-4a6692013f67.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMzNjc1MTQsIm5iZiI6MTc1MzM2NzIxNCwicGF0aCI6Ii84Nzg2NDQzOS80NzAzMzU0ODEtMzlhYmI0MDktMzY2ZS00MTNlLWI3MWQtNGE2NjkyMDEzZjY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzI0VDE0MjY1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5NmM4NzUzNzVlMTY4Njk3YzlkMWY0Zjg5ZjRmMWIzOWNmMTM5Y2M2OWU0NDNkODVhYWU3MTIyZDVlMDFkNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XKcfn917NLTDOFQf7nxvyaWGo9dDHY2-8eWLuWWVIKk">
> > > but I am not sure why the connection is refusing:
> > > ```
> > > 2025-07-24 13:47:00.932 [debu]  apphealth: error checking app health  id=9830f5f2-bf28-4ef1-981a-090ccaf65dbb  slug=pgadmin  now_unhealthy=true  error="Get \"http://localhost:5050/\": dial tcp [::1]:5050: connect: connection refused"
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > @DevelopmentCats Did you encounter anything like this during your local testing? Let me know if you have any suggestions for resolving it.
> > 
> > 
> > Even in my version, it still took forever to build.
> > PGAdmin seems to be pretty heavy overall, and I doubt there is much to be done to speed the install up. In my testing though I was able to get pgadmin installed, and available over the coder app button.
> 
> Were you able to open it?

It was not loading up when I pulled and tested it last, but let me build and test again. Maybe I can provide you some valuable output

AJ0070 commented:
@DevelopmentCats added logo on coder: https://github.com/coder/coder/pull/19210

AJ0070 commented:
> It seems to be loading fine for me when I just pulled it. @AJ0070
> 
> Can you upload the icon for this and pull the svg for pgadmin instead of postgres like its referencing right now. Make sure to also make a PR in coder/coder to upload the icon in site/static/icons so that it shows up properly in coder.
> 
> Also there is a typo in your avatar image please fix that.
> 
> Lastly I notice that you have changes in here for the filebrowser module. Please remove that. I will review this after you take care of those things

@DevelopmentCats Ready for review

matifali commented:
@AJ0070 Run `bun fmt` and commit
